server/datastore/mysql/apple_mdm.go

```diff
@@ -4789,3 +4801,184 @@ WHERE
 		tok.ID)
 	return ctxerr.Wrap(ctx, err, "updating abm_token")
 }
+
+func (ds *Datastore) InsertABMToken(ctx context.Context, tok *fleet.ABMToken) (*fleet.ABMToken, error) {
+	const stmt = `
+INSERT INTO
+	abm_tokens
+	(organization_name, apple_id, terms_expired, renew_at, token, macos_default_team_id, ios_default_team_id, ipados_default_team_id)
+VALUES (?, ?, ?, ?, ?, ?, ?, ?)
+ON DUPLICATE KEY UPDATE
```

Comment:
I'm not sure we want that here, because it means we will never have duplicate org name errors - we will always overwrite the token with the same org name. I think we need to be explicit - we either try to insert and fail because the org name already exists (to be used when a new token is uploaded), or we update and fail if the to-be-updated token ID does NOT exist (to be used when renewing/updating teams of an existing token).